web/src/stream_settings_ui.js

```diff
@@ -420,6 +424,13 @@ export function update_empty_left_panel_message() {
     } else {
         has_streams = stream_data.get_unsorted_subs().length;
     }
+    // Show "no channels match" text if all channels are hidden and there's a search query.
+    if (hidden_ids.size === all_stream_ids.length && left_panel_params.input) {
```

Comment:
I think would be better to just check whether any streams are displayed using Jquery instead of passing these parameters to the function.